{"result_id": "0d73e4754b384af9bb180f02bd378ac2", "data_type": null, "error": {"code": "from typing import Dict, List, Union\n\nimport numpy as np\nfrom elasticsearch import Elasticsearch\n\nif 'data_loader' not in globals():\n    from mage_ai.data_preparation.decorators import data_loader\nif 'test' not in globals():\n    from mage_ai.data_preparation.decorators import test\n\n\n@data_loader\ndef search(query_embedding: Union[List[int], np.ndarray], *args, **kwargs) -> List[Dict]:\n    connection_string = kwargs.get('connection_string', 'http://localhost:9200')\n    index_name = kwargs.get('index_name', 'documents')\n    source = kwargs.get('source', \"cosineSimilarity(params.query_vector, 'embedding') + 1.0\")\n    top_k = kwargs.get('top_k', 5)\n    chunk_column = kwargs.get('chunk_column', 'content')\n\n    if isinstance(query_embedding, np.ndarray):\n        query_embedding = query_embedding.tolist()\n\n    script_query = dict(\n        script_score=dict(\n            query=dict(match_all=dict()),\n            script=dict(source=source, params=dict(query_vector=query_embedding)),\n        )\n    )\n\n    es_client = Elasticsearch(connection_string)\n\n    response = es_client.search(\n        index=index_name,\n        query=dict(\n            size=top_k,\n            query=script_query,\n            _source=[chunk_column],\n        ),\n    )\n\n    return [hit['_source']['content'] for hit in response['hits']['hits']]\n\n\n@test\ndef test_output(output, *args) -> None:\n    \"\"\"\n    Template code for testing the output of the block.\n    \"\"\"\n    assert output is not None, 'The output is undefined'", "code_context": [], "code_context_formatted": [], "error": "Block infinite_photon may be missing upstream dependencies. It expected to have at least 1 arguments, but only received 0. Confirm that the @data_loader method declaration has the correct number of arguments.", "errors": null, "exception": "Exception: Block infinite_photon may be missing upstream dependencies. It expected to have at least 1 arguments, but only received 0. Confirm that the @data_loader method declaration has the correct number of arguments.", "line_number": 1766, "message": "Block infinite_photon may be missing upstream dependencies. It expected to have at least 1 arguments, but only received 0. Confirm that the @data_loader method declaration has the correct number of arguments.", "message_formatted": "Block infinite_photon may be missing upstream dependencies. It expected to have at least 1 arguments, but only received 0. Confirm that the @data_loader method declaration has the correct number of arguments.", "stacktrace": ["Traceback (most recent call last):\n", "  File \"/usr/local/lib/python3.10/site-packages/mage_ai/kernels/magic/execution.py\", line 281, in execute_code_async\n    local_variables = await __modify_and_execute(\n", "  File \"/usr/local/lib/python3.10/site-packages/mage_ai/kernels/magic/execution.py\", line 135, in __modify_and_execute\n    raise error\n", "  File \"/usr/local/lib/python3.10/site-packages/mage_ai/kernels/magic/execution.py\", line 117, in __modify_and_execute\n    res = await res\n", "  File \"/usr/local/lib/python3.10/site-packages/mage_ai/kernels/magic/environments/setup_helpers.py\", line 48, in execute\n    await block.execute(\n", "  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\", line 1711, in execute\n    await loop.run_in_executor(\n", "  File \"/usr/local/lib/python3.10/concurrent/futures/thread.py\", line 58, in run\n    result = self.fn(*self.args, **self.kwargs)\n", "  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\", line 1688, in execute_sync\n    return __execute()\n", "  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\", line 1665, in __execute\n    raise err\n", "  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\", line 1567, in __execute\n    output = self.execute_block(\n", "  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\", line 1912, in execute_block\n    outputs = self._execute_block(\n", "  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\", line 2063, in _execute_block\n    block_function = self._validate_execution(decorated_functions, input_vars)\n", "  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\", line 1766, in _validate_execution\n    raise Exception(\n", "Exception: Block infinite_photon may be missing upstream dependencies. It expected to have at least 1 arguments, but only received 0. Confirm that the @data_loader method declaration has the correct number of arguments.\n"], "stacktrace_formatted": ["\u001b[91mTraceback (most recent call last):\n\u001b[0m", "\u001b[34m  File \"/usr/local/lib/python3.10/site-packages/mage_ai/kernels/magic/execution.py\"\u001b[0m, \u001b[36mline 281\u001b[0m, \u001b[35min execute_code_async\n    local_variables = await __modify_and_execute(\n\u001b[0m", "\u001b[34m  File \"/usr/local/lib/python3.10/site-packages/mage_ai/kernels/magic/execution.py\"\u001b[0m, \u001b[36mline 135\u001b[0m, \u001b[35min __modify_and_execute\n    raise error\n\u001b[0m", "\u001b[34m  File \"/usr/local/lib/python3.10/site-packages/mage_ai/kernels/magic/execution.py\"\u001b[0m, \u001b[36mline 117\u001b[0m, \u001b[35min __modify_and_execute\n    res = await res\n\u001b[0m", "\u001b[34m  File \"/usr/local/lib/python3.10/site-packages/mage_ai/kernels/magic/environments/setup_helpers.py\"\u001b[0m, \u001b[36mline 48\u001b[0m, \u001b[35min execute\n    await block.execute(\n\u001b[0m", "\u001b[34m  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\"\u001b[0m, \u001b[36mline 1711\u001b[0m, \u001b[35min execute\n    await loop.run_in_executor(\n\u001b[0m", "\u001b[34m  File \"/usr/local/lib/python3.10/concurrent/futures/thread.py\"\u001b[0m, \u001b[36mline 58\u001b[0m, \u001b[35min run\n    result = self.fn(*self.args, **self.kwargs)\n\u001b[0m", "\u001b[34m  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\"\u001b[0m, \u001b[36mline 1688\u001b[0m, \u001b[35min execute_sync\n    return __execute()\n\u001b[0m", "\u001b[34m  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\"\u001b[0m, \u001b[36mline 1665\u001b[0m, \u001b[35min __execute\n    raise err\n\u001b[0m", "\u001b[34m  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\"\u001b[0m, \u001b[36mline 1567\u001b[0m, \u001b[35min __execute\n    output = self.execute_block(\n\u001b[0m", "\u001b[34m  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\"\u001b[0m, \u001b[36mline 1912\u001b[0m, \u001b[35min execute_block\n    outputs = self._execute_block(\n\u001b[0m", "\u001b[34m  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\"\u001b[0m, \u001b[36mline 2063\u001b[0m, \u001b[35min _execute_block\n    block_function = self._validate_execution(decorated_functions, input_vars)\n\u001b[0m", "\u001b[34m  File \"/usr/local/lib/python3.10/site-packages/mage_ai/data_preparation/models/block/__init__.py\"\u001b[0m, \u001b[36mline 1766\u001b[0m, \u001b[35min _validate_execution\n    raise Exception(\n\u001b[0m", "\u001b[90mException: Block infinite_photon may be missing upstream dependencies. It expected to have at least 1 arguments, but only received 0. Confirm that the @data_loader method declaration has the correct number of arguments.\n\u001b[0m", "\u001b[91mBlock infinite_photon may be missing upstream dependencies. It expected to have at least 1 arguments, but only received 0. Confirm that the @data_loader method declaration has the correct number of arguments.\u001b[0m"], "type": "Exception"}, "metadata": null, "output": null, "process": {"data": [], "exitcode": null, "is_alive": false, "internal_state": "INIT", "kernel_uuid": "infinite_photon", "message": "from typing import Dict, List, Union\n\nimport numpy as np\nfrom elasticsearch import Elasticsearch\n\nif 'data_loader' not in globals():\n    from mage_ai.data_preparation.decorators import data_loader\nif 'test' not in globals():\n    from mage_ai.data_preparation.decorators import test\n\n\n@data_loader\ndef search(query_embedding: Union[List[int], np.ndarray], *args, **kwargs) -> List[Dict]:\n    connection_string = kwargs.get('connection_string', 'http://localhost:9200')\n    index_name = kwargs.get('index_name', 'documents')\n    source = kwargs.get('source', \"cosineSimilarity(params.query_vector, 'embedding') + 1.0\")\n    top_k = kwargs.get('top_k', 5)\n    chunk_column = kwargs.get('chunk_column', 'content')\n\n    if isinstance(query_embedding, np.ndarray):\n        query_embedding = query_embedding.tolist()\n\n    script_query = dict(\n        script_score=dict(\n            query=dict(match_all=dict()),\n            script=dict(source=source, params=dict(query_vector=query_embedding)),\n        )\n    )\n\n    es_client = Elasticsearch(connection_string)\n\n    response = es_client.search(\n        index=index_name,\n        query=dict(\n            size=top_k,\n            query=script_query,\n            _source=[chunk_column],\n        ),\n    )\n\n    return [hit['_source']['content'] for hit in response['hits']['hits']]\n\n\n@test\ndef test_output(output, *args) -> None:\n    \"\"\"\n    Template code for testing the output of the block.\n    \"\"\"\n    assert output is not None, 'The output is undefined'", "message_request_uuid": "1724235497328", "message_uuid": "f2ecf008c4a0475c9507bd681fc97146", "output_manager": {"namespace": "pipeline/illuminating_zenith", "path": "llm/rager/data_loaders/infinite_photon.py", "uuid": "1724235497328"}, "pid": "f2ecf008c4a0475c9507bd681fc97146", "pid_spawn": null, "source": "infinite_photon", "stream": "code_executions", "timestamp": null, "timestamp_end": null, "uuid": "infinite_photon"}, "status": "error", "type": "status", "uuid": "infinite_photon", "timestamp": 1724235503524, "output_text": null}
{"result_id": "584c4d4c691f41c181f5a66c71e7f413", "data_type": null, "error": null, "metadata": {"namespace": "pipeline/illuminating_zenith", "path": "llm/rager/data_loaders/infinite_photon.py", "uuid": "1724235497328", "block_type": "data_loader", "block_uuid": "infinite_photon", "execution_partition": null, "pipeline_uuid": "illuminating_zenith"}, "output": "", "process": {"data": [], "exitcode": null, "is_alive": false, "internal_state": "INIT", "kernel_uuid": "infinite_photon", "message": "from typing import Dict, List, Union\n\nimport numpy as np\nfrom elasticsearch import Elasticsearch\n\nif 'data_loader' not in globals():\n    from mage_ai.data_preparation.decorators import data_loader\nif 'test' not in globals():\n    from mage_ai.data_preparation.decorators import test\n\n\n@data_loader\ndef search(query_embedding: Union[List[int], np.ndarray], *args, **kwargs) -> List[Dict]:\n    connection_string = kwargs.get('connection_string', 'http://localhost:9200')\n    index_name = kwargs.get('index_name', 'documents')\n    source = kwargs.get('source', \"cosineSimilarity(params.query_vector, 'embedding') + 1.0\")\n    top_k = kwargs.get('top_k', 5)\n    chunk_column = kwargs.get('chunk_column', 'content')\n\n    if isinstance(query_embedding, np.ndarray):\n        query_embedding = query_embedding.tolist()\n\n    script_query = dict(\n        script_score=dict(\n            query=dict(match_all=dict()),\n            script=dict(source=source, params=dict(query_vector=query_embedding)),\n        )\n    )\n\n    es_client = Elasticsearch(connection_string)\n\n    response = es_client.search(\n        index=index_name,\n        query=dict(\n            size=top_k,\n            query=script_query,\n            _source=[chunk_column],\n        ),\n    )\n\n    return [hit['_source']['content'] for hit in response['hits']['hits']]\n\n\n@test\ndef test_output(output, *args) -> None:\n    \"\"\"\n    Template code for testing the output of the block.\n    \"\"\"\n    assert output is not None, 'The output is undefined'", "message_request_uuid": "1724235497328", "message_uuid": "f2ecf008c4a0475c9507bd681fc97146", "output_manager": {"namespace": "pipeline/illuminating_zenith", "path": "llm/rager/data_loaders/infinite_photon.py", "uuid": "1724235497328"}, "pid": "f2ecf008c4a0475c9507bd681fc97146", "pid_spawn": null, "source": "infinite_photon", "stream": "code_executions", "timestamp": null, "timestamp_end": null, "uuid": "infinite_photon"}, "status": "success", "type": "output", "uuid": "infinite_photon", "timestamp": 1724235503566, "output_text": ""}
